### PR TITLE
Migration guide for `ButtonBar` deprecation

### PR DESCRIPTION
**Pull request by TahaTesser**
Related to https://github.com/flutter/flutter/pull/145523

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

**Review by Piinks — COMMENTED** [review]
Ack, let's not merge until the PR lands in the framework. I had a left-field question about possibly refactoring ButtonBar.

**Comment by TahaTesser** [comment]
Marked this as ready to review since https://github.com/flutter/flutter/pull/145523 landed.

File: src/release/breaking-changes/deprecate-buttonbar.md
Comment (Piinks) [inline src/release/breaking-changes/deprecate-buttonbar.md:10]:
```suggestion
The `ButtonBar` widget was deprecated in favor of the more efficient
```

File: src/release/breaking-changes/deprecate-buttonbar.md
Comment (Piinks) [inline src/release/breaking-changes/deprecate-buttonbar.md:12]:
```suggestion
`ButtonBarTheme` were also deprecated.
```

File: src/release/breaking-changes/deprecate-buttonbar.md
Comment (Piinks) [inline src/release/breaking-changes/deprecate-buttonbar.md:16]:
```suggestion
The `ButtonBar` widget lays out its children in a row and in a column if there is
```

File: src/release/breaking-changes/deprecate-buttonbar.md
Comment (Piinks) [inline src/release/breaking-changes/deprecate-buttonbar.md:29]:
```suggestion
  between buttons. Wrap the `OverflowBar` widget with `Padding` widget to
```

File: src/release/breaking-changes/deprecate-buttonbar.md
Comment (Piinks) [inline src/release/breaking-changes/deprecate-buttonbar.md:34]:
```suggestion
- Remove `ButtonBarThemeData` from `ThemeData`, if it exists.
```

**Review by Piinks — APPROVED** [review]
LGTM!

**Pull request merged**
